### PR TITLE
Add finalize functionality for indices #9 #10 #11

### LINKED ISSUE #9: A user should be able to edit an index that has not been finalized.
A user should be able to edit an index that has not been finalized.

### LINKED ISSUE #10: A user should be able to finalize an index they have entered.
A user should be able to finalize an index they have entered.

### LINKED ISSUE #11: A user should be able to delete an index that has not been finalized.
A user should be able to delete an index that has not been finalized.

### PR DESCRIPTION
**Pull request by grimm**
Fixes #9 
Fixes #10 
Fixes #11

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -4,7 +4,9 @@
   resources :choices, except: :destroy
   resources :categories, except: :destroy
   resources :communities, shallow: true do
-    resources :indices
+    resources :indices do
+      get :finalize, on: :member
```
Comment (teknofire) [inline config/routes.rb]: if this is changing anything about the index this should be a patch operation instead of get

**Comment by grimm** [comment]
Ready for review.

**Comment by teknofire** [comment]
Need to make sure the ability tests get wrapped up into their own class instead of being in the module

**Comment by grimm** [comment]
Ready to review

**Comment by grimm** [comment]
Ready for review

**Comment by teknofire** [comment]
:+1:

**Comment by teknofire** [comment]
@smacfarlane no notice was sent out but this should be ready to review.  Want to get this merged in so we I can get some additional work lined up for cleaning up the community pages and workflows

**Comment by smacfarlane** [comment]
I'm not super happy with the indices controller test,  we've got two indices being loaded in setup that are essentially the same thing.  This is my fault for not removing the finalized_at field in the {incomplete, partial_complete, complete} indices fixtures.  I think we should get rid of the unfinalized fixture, fix the above three, and just load one index in setup. 

Also the, destroy/finalize confirms should be more specific about what they're actually confirming:
- This will permanently delete this Index,  are you sure?
- You will no longer be able to make changes to this index once finalized, are you sure? 

Finally, the {finalize,edit} buttons margin-top changes from 5px to 1px on mouseover.  Not sure what might be causing that.

**Comment by teknofire** [comment]
The button margin issue is a problem with the custom bootstrap theme, I'm actually working fixing that as part of the main page changes

**Comment by grimm** [comment]
@gina-alaska/awrvi not quite ready sorry.

File: test/controllers/indices_controller_test.rb [hunk 2]
```diff
@@ -77,5 +77,9 @@ class IndicesControllerTest < ActionDispatch::IntegrationTest
     login_as(users(:two))
     patch finalize_index_url(@index)
     assert_redirected_to index_path(@index)
+
+    assert_difference('Index.count', 0) do
```
Comment (teknofire) [inline test/controllers/indices_controller_test.rb:85]: This should really be in a separate test for not allowing deletion of a finalized item.  In which case selecting a different index item for just this test would be fine.

**Comment by grimm** [comment]
@gina-alaska/awrvi  Ready for review.

**Comment by grimm** [comment]
@gina-alaska/awrvi  Ready for review.

**Comment by teknofire** [comment]
Looking good!

**Comment by smacfarlane** [comment]
:+1:

**Pull request merged**